indico/modules/auth/controllers.py

```diff
@@ -253,20 +289,30 @@ def _process(self):
                 value = getattr(pending, field.short_name, '')
                 if value and not field.data:
                     field.data = value
+
         if pending:
             flash(_("There is already some information in Indico that concerns you. "
                     "We are going to link it automatically."), 'info')
         return WPAuth.render_template('register.html', form=form, local=(not self.identity_info),
                                       must_verify_email=handler.must_verify_email, widget_attrs=handler.widget_attrs,
-                                      email_sent=session.pop('register_verification_email_sent', False))
+                                      email_sent=session.pop('register_verification_email_sent', False),
+                                      approval_email_sent=session.pop('approval_email_sent', False))
 
-    def _send_confirmation(self, email):
+    def _send_approval_mail_to_admin(self, new_account_email):
+        session['approval_email_sent'] = True
+        return _send_approval_mail(new_account_email)
+
+    def _send_confirmation(self, email, flashed_msg=None):
         session['register_verification_email_sent'] = True
         return _send_confirmation(email, 'register-email', '.register', 'auth/emails/register_verify_email.txt',
-                                  url_args={'provider': self.provider_name})
+                                  url_args={'provider': self.provider_name}, flashed_msg=flashed_msg)
 
     def _create_user(self, form, handler, pending_user):
         data = form.data
+
+        if session.get('approval_email_sent', False):
```

Comment:
i think you could unconditionally delete it here..?

    session.pop('approval_email_sent', None)